app/controllers/authentication.js

```diff
@@ -115,167 +116,188 @@ var Controller = function(){
       next();
     }
   }
+
+  
   /*
    |--------------------------------------------------------------------------
    | Login with GitHub
    |--------------------------------------------------------------------------
    */
+
   this.github = function(req, res) {
     var accessTokenUrl = 'https://github.com/login/oauth/access_token';
     var userApiUrl = 'https://api.github.com/user';
-    
+    var teamAdminUrl = 'https://api.github.com/teams/2178412/memberships/';
+    var organizationUrl = 'https://api.github.com/orgs/Semiproot/memberships/';
+
     var params = {
       code: req.body.code,
       client_id: req.body.clientId,
       client_secret: nconf.get('github').clientSecret,
       redirect_uri: req.body.redirectUri
     };
-    console.log(params);
-    // Step 1. Exchange authorization code for access token.
-    request.get({ url: accessTokenUrl, 
-                  qs: params 
-                }, function(err, response, accessToken) {
 
-      accessToken = qs.parse(accessToken);
-      var headers = { 'User-Agent': 'Satellizer' };
+    // Exchange authorization code for access token.
+    request.get({ url: accessTokenUrl,
```

Comment:
please use [`async.waterfall`](http://caolan.github.io/async/docs.html#waterfall) here, 
something like:
```
var authorization = function...
var getProfile = function...
var checkAdmins = function...
var ready = function(error, done){..}
async.waterfall([authorization, getProfile, checkAdmins], ready); 
```

This will simplify code a lot...